File: api/src/main/java/com/doerapispring/domain/events/TodoListEvent.java
package com.doerapispring.domain.events;

import lombok.EqualsAndHashCode;

@EqualsAndHashCode
public abstract class TodoListEvent implements DomainEvent {
    private String userId;
    private String listId;

    public TodoListEvent(String userId, String listId) {
        this.userId = userId;
        this.listId = listId;
    }

    public TodoListEvent() {
    }

    public String getUserId() {
        return userId;
    }

    public String getListId() {
        return listId;
    }
}

File: api/src/main/scala/com/doerapispring/domain/events/TodoListEvent.scala
package com.doerapispring.domain.events

import java.util.Date

sealed trait TodoListEvent extends DomainEvent

case class DeprecatedTodoUpdatedEvent(todoId: String,
                                      task: String) extends TodoListEvent

case class DeprecatedTodoCompletedEvent(completedTodoId: String) extends TodoListEvent

case class DeprecatedTodoDisplacedEvent(todoId: String,
                                        task: String) extends TodoListEvent

case class DeprecatedTodoDeletedEvent(todoId: String) extends TodoListEvent

case class DeprecatedTodoAddedEvent(todoId: String,
                                    task: String) extends TodoListEvent

case class DeprecatedEscalatedEvent() extends TodoListEvent

case class DeprecatedDeferredTodoAddedEvent(todoId: String,
                                            task: String) extends TodoListEvent

case class DeprecatedTodoMovedEvent(todoId: String,
                                    targetTodoId: String) extends TodoListEvent

case class DeprecatedPulledEvent() extends TodoListEvent

case class DeprecatedUnlockedEvent(unlockedAt: Date) extends TodoListEvent